PrincetonUniversity/PsyNeuLink
# v0.4.8.0

Mechanisms
* Add ContrastiveHebbianMechanism (but ContrastiveHebbian Function is not yet implemented)
* Add condition attribute; automatically assigned to scheduler of any System to which Mechanism belongs
* Add reinitialize_when Condition on Mechanisms that accumulate 

Functions
* Add ReLU (rectified linear transformation) Function
* Distance Function:  added MAX_ABS_DIFF metric

Compositions
* Add pre-alpha/incomplete Composition
    * intended to be more flexible than and eventually replace System/Process

* System
    * Add runtime parameters to Run
        * Alternative parameter values that are used when their Condition is True 
    * show_graph:  added show_process option to organize by Processes

* Process
   * Add warning for (and ignore) duplicate projection specification

Models
* Add PCTC (Kalanthroff et al., 2018)
* Add Stroop GRAIN (Cohen & Huston, 1994)
* Add Conflict Monitoring and Cognitive Control (Botvinick et al., 2001)

Scheduling
* Add Time-based Conditions for TimeScale.TIME_STEP

Testing
* Add testing on Travis (OS X, Linux)
* Add testing on Appveyor (Windows)
* Add coverage testing (pytest-cov, Coveralls)

Misc
* Add support for python3.7
* Many assorted bug fixes and improvements (see commit log for full list)
* Assorted documentation improvements
